enumerable.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+# rubocop:disable all
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield self[i]
+        i += 1
+      end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+    array2 = []
+      i = 0
+      while i < size
+        array2 << self[i] if yield self[i]
+        i += 1
+      end
+    array2
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif arg.is_a? Regexp
+      my_each { |element| return false unless arg =~ element }
+    elsif arg.is_a? Class
+      my_each { |element| return true unless element.class == arg }
+    else
+      my_each { |element| return false unless element }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |element| return true if yield(element) }
+    elsif arg.is_a? Regexp
+      my_each { |element| return true if arg =~ element }
+    elsif arg.is_a? Class
+      my_each { |element| return true if element.class == arg }
+    else
+      my_each { |element| return true if element }
+    end
+    false
+  end
```

Comment:
* [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_any?(5) == array.any?(5) # false
  ```